chainer/functions/pooling/roi_max_pooling_2d.py

```diff
@@ -0,0 +1,306 @@
+# Modified work:
+# -----------------------------------------------------------------------------
+# Copyright (c) 2015 Preferred Infrastructure, Inc.
+# Copyright (c) 2015 Preferred Networks, Inc.
+# -----------------------------------------------------------------------------
+
+# Original work of _roi_pooling_slice, forward_cpu and backward_cpu:
+# -----------------------------------------------------------------------------
+# Copyright 2014 Nervana Systems Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+# Original work of forward_gpu and backward_gpu:
+# -----------------------------------------------------------------------------
+# Fast R-CNN
+# Copyright (c) 2015 Microsoft
+# Licensed under The MIT License [see fast-rcnn/LICENSE for details]
+# Written by Ross Girshick
+# -----------------------------------------------------------------------------
+
+import numpy
+import six
+
+import chainer
+from chainer.backends import cuda
+from chainer import function
+from chainer.utils import type_check
+
+from chainer.functions.pooling.roi_pooling_2d import _roi_pooling_slice
+
+
+def _pair(x):
+    if isinstance(x, chainer.utils.collections_abc.Iterable):
+        return x
+    return x, x
+
+
+class ROIMaxPooling2D(function.Function):
+
+    """RoI max pooling over a set of 2d planes."""
+
+    def __init__(self, outsize, spatial_scale):
+        outh, outw = _pair(outsize)
+        if not (isinstance(outh, int) and outh > 0):
+            raise TypeError(
+                'outsize[0] must be positive integer: {}, {}'
+                .format(type(outh), outh))
+        if not (isinstance(outw, int) and outw > 0):
+            raise TypeError(
+                'outsize[1] must be positive integer: {}, {}'
+                .format(type(outw), outw))
+        if isinstance(spatial_scale, int):
```

Comment:
This check accepts negative int (e.g. `-1`). In addition, it cannot take number-like types (e.g. 'numpy.float32').
How about this?
```python
try:
    spatial_scale = float(spatial_scale)
except ValueError:
    # raise an error for casting
if spatial_scale <= 0:
    # raise an error for non-positive value
```

Reply:
> In addition, it cannot take number-like types (e.g. 'numpy.float32').

We discussed this problem locally and decided not to support `numpy.scalar` in this PR.

Reply:
@Hakuyume BTW, i found that `numpy.float64` works.

```python
Python 2.7.12 (default, Dec  4 2017, 14:50:18) 
Type "copyright", "credits" or "license" for more information.

IPython 5.5.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import numpy as np

In [2]: a = np.float64(1.6)

In [3]: isinstance(a, float)
Out[3]: True

In [4]: b = np.float32(1.6)

In [5]: isinstance(b, float)
Out[5]: False

```

Reply:
:tada:

Reply:
`np.issubtype` seems good for checking scalar value.
https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.issubdtype.html

Reply:
It raises a warning for `float`.
```python
>>> np.issubdtype(type(1.0), float)
/usr/lib/python3.7/site-packages/ipykernel_launcher.py:1: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
```

Reply:
how about this?
```python
np.issubdtype(type(1.0), np.floating)
# no warnings
np.issubdtype(type(1), np.integer)
# no warnings
```